### DISCUSSION TITLE
Attempting to create server using API causes Error 422: Unprocessable Entity.

### DISCUSSION THREAD

**Discussion by RonaldBunk** [pterodactyl/panel#2902] [Support]
Hello,

I am attemping to create a server using the Pterodactyl API however when I send the request with the following payload, it responds with an **Error 422: Unprocessable Entity**:
```json
{
   "name": "Realm",
   "user": 1,
   "egg": 3,
   "docker_image": "quay.io/pterodactyl/core:java",
   "startup": "java -Xms128M -Xmx512M -jar server.jar",
   "environment": {
      "MINECRAFT_VERSION": "latest",
      "SERVER_JARFILE": "server.jar"
   },
   "limits": {
      "memory": 512,
      "swap": 0,
      "disk": 2048,
      "io": 500,
      "cpu": 100
   },
   "feature_limits": {
      "databases": 1,
      "backups": 5
   },
   "allocation": {
      "default": 17
   }
}
```

**Edited the payload, not sure if that's any better; does not work either:**
```json
{
   "name": "REALM-CONSOLE",
   "user_id": 1,
   "egg": 3,
   "startup": "java -Xms128M -Xmx512M",
   "docker_image": "quay.io/pterodactyl/core:java",
   "external_id": "REALM-CONSOLE",
   "environment": {
      "MINECRAFT_VERSION": "1.16.4",
      "BUILD_NUMBER": "latest",
      "SERVER_JARFILE": "server.jar"
   },
   "feature_limits": {
      "databases": 1,
      "backups": 5,
      "allocations": 2
   },
   "limits": {
      "memory": 512,
      "swap": 0,
      "disk": 2048,
      "io": 500,
      "cpu": 100
   },
   "deploy": {
      "port_range": [
         "10000-10100"
      ],
      "locations": [
         1
      ],
      "dedicated_ip": false
   }
}
```

I am using https://dashflo.net/docs/api/pterodactyl/v1/ as reference.

**Comment by opiepl** [comment]
Do you have a ptero on https protocol and not on http?

**Reply by RonaldBunk** [reply]
In my case HTTP, but the executing code does allow for HTTPS, why?

**Reply by opiepl** [reply]
Ptero does not allow using API while the panel is on http

**Reply by RonaldBunk** [reply]
That seems rather peculiar, as other HTTP requests (Like user creation) do work. 
Besides, the response message does not really signal that it would not tolerate any traffic it primarily states that something is wrong with the request itself.

Which I have also seen with other requests, when the payload wasn't correct. Except I don't know the correct payload for server creation.

**Comment by DaneEveritt** [comment]
Whats the exact error response from the Panel. The error code is only slightly useful, it should be telling you exactly what the validation issue is in the response body.

**Comment by SyedaFareed** [comment]
Dealing with the same error. Anyone solved this yet?

**Comment by MrTrinax** [comment]
This reply is rather late, but for anyone stumbling through here asking the same question, make sure your body contains `allocation`, as this is required.

As the example show:
`"allocation": {
      "default": 17
   }`

`allocation.default`, which is the ID for a IP/PORT allocation on the node, can be obtain by making a GET request to your node at "/nodes/{node_id}/allocations".

---

Now for why I ended up here. I wonder if I'm supposed to make a GET request for available allocations to get the ID each time a new server is to be created, or if I can do it in a more efficient way. Could it be possible to specify ports instead of IDs?
